File: c++/CommandPattern/CookBeefCommand.cpp
#include "stdafx.h"
#include "CookBeefCommand.h"


CookBeefCommand::CookBeefCommand(Chef chef)
{
	this->chef = chef;
	Command::Name = "±¬³´Å£Èâ";
}


void CookBeefCommand::Cook()
{
	this->chef.CookBeef();
}

File: csharp/DesignPattern.Command/CookBeefCommand.cs

namespace DesignPattern.Command
{
    /// <summary>
    /// 爆炒牛肉
    /// </summary>
    public class CookBeefCommand : Command
    {
        private Chef chef;

        public CookBeefCommand(Chef chef)
        {
            this.chef = chef;
            base.Name = "爆炒牛肉";
        }

        public override void Cook()
        {
            this.chef.CookBeef();
        }


    }
}
